tls/s2n_handshake_io.c

```diff
@@ -1348,6 +1348,7 @@ static int s2n_handle_retry_state(struct s2n_connection *conn)
     } else {
         /* The read handler processed the record successfully, we are done with this
          * record. Advance the state machine. */
+        POSIX_GUARD(s2n_tls13_handle_secrets(conn));
         POSIX_GUARD(s2n_advance_message(conn));
     }
```

Comment:
Should `s2n_tls13_handle_secrets` be called from `s2n_advance_message` instead?

Reply:
It probably should so that we don't make this mistake again, but that turns into a bit of a more involved change because the SAW proofs will need to be updated. I can create a separate issue though.

Reply:
https://github.com/aws/s2n-tls/issues/2724